deploy/setup.sh

```diff
@@ -2,85 +2,38 @@
 #All constant variables are in upper case convention. They are:
 #  INSTALL_DIR : path to the installation directory
 #  MAIN_SERVER_PUBLIC : path to the main_server/public directory inside the installation directory
-#All non constant variables are in lower case convention. They are:
-#  user : user running the script
-sudo true
-
-#install docker dependecy for AUFS file system
-sudo apt-get install -y lxc wget bsdtar curl
-sudo apt-get install -y "linux-image-extra-$(uname -r)"
-sudo modprobe aufs
-wget -qO- https://get.docker.com/ | sh
-
-sudo apt update
-sudo apt install -y python-pip libssl-dev sshpass libffi-dev build-essential python-dev
-sudo pip install --upgrade pip
-sudo pip install cryptography
-sudo pip install setuptools
-sudo pip install ansible
-sudo service docker restart
-
-INSTALL_DIR="/opt/autolabjs"
-sudo mkdir -p "$INSTALL_DIR"
-user="$(whoami)"
-sudo chown -R "$user" "$INSTALL_DIR"
-#The below command can be uncommented if docker privileges are to be
-#granted to the current user. This is generally not advisable.
-#sudo usermod -aG docker "$user"
-
-mkdir -p "$INSTALL_DIR"/gitlab
-mkdir -p "$INSTALL_DIR"/gitlab/config
-mkdir -p "$INSTALL_DIR"/gitlab/logs
-mkdir -p "$INSTALL_DIR"/gitlab/data
-mkdir -p "$INSTALL_DIR"/mysql
-
-mkdir -p "$INSTALL_DIR"/main_server
-cp -rf ../main_server/* "$INSTALL_DIR"/main_server/
-npm install --prefix "$INSTALL_DIR"/main_server
+#  USER : user running the script.
+# All these variables are defined in the config file at the location specified by the
+# variable CONFIG
+
+CONFIG=./setup.conf
+if [[ -f $CONFIG ]]
+then
+  # shellcheck disable=SC1090
+  . "$CONFIG"
+fi
+
```

Comment:
node and npm need to be installed before they can be used. The commands are:
```
curl -sL https://deb.nodesource.com/setup_8.x -o nodesource_setup.sh
sudo bash nodesource_setup.sh
sudo apt-get update && sudo apt-get install --force-yes -y nodejs build-essential git
sudo rm nodesource_setup.sh
```

Reply:
The dependencies file is already run before. That installs nodejs and npm before this script is run.

Reply:
Is `git` required to be installed in this case?